controllers/policyserver_controller.go

```diff
@@ -80,6 +90,12 @@ func (r *PolicyServerReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	reconcileResult, reconcileErr := r.reconcile(ctx, &policyServer, policies)
 
 	if err := r.Client.Status().Update(ctx, &policyServer); err != nil {
+		if apierrors.IsConflict(err) {
```

Comment:
Why can we run into a conflict now? Is that because the policy server could be changed by the new CA reconciliation loop too?

Reply:
Yes, the CA controller also updates policy server by setting the conditions to the policy server about secret reconciliation

Reply:
Let me share some toughs that I had while I was coding this controller and see what you think. I believe this is pertinent for this discussion.  

In this PR version the policy server controller still takes care of the secret of the policy server. It will recreate it if it's missing. And the CA controller will touch policy server secret only when it creates the root CA. Which leads to too controller updating policy server resource. The more I think about this more I'm in favor to remove completely the policy server CA reconciliation from the policy server controller and move it to the CA controller. 

I've decided to keep policy server CA reconciliation in the policy server controller as well because it sounds nice to have "all policy server stuff" in a single controller. This will not fix the conflicts on updates. But the code may be simpler to maintain if we move all the CA secret reconciliation to the dedicated controller and change the policy server controller to just wait for the secret to show up before continuing the reconciliation. 

What do you think @kubewarden/kubewarden-developers